File: minwook/6주차/추억_점수.java
import java.util.HashMap;
class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        HashMap<String, Integer> map = new HashMap<>();
        for(int n=0; n<name.length; n++){
            map.put(name[n],yearning[n]);
        }
        for(int i=0; i<photo.length; i++){
            for(String j : photo[i]){
                if(map.containsKey(j))
                    answer[i] += map.get(j);
            }
        }
        return answer;
    }
}

File: yeongseong/6week/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    let score = 0;
    for (let i = 0; i < photo.length; i++){
        score = 0;
        for (let j = 0; j < name.length; j++){
            if (photo[i].includes(name[j])) {
                score += yearning[j];
            }
        }
        answer.push(score);
    }
    return answer;
}
